app/Svg/Parser.hs

```diff
@@ -255,27 +262,28 @@ parsePathHelper key trans pathTag =
                 ""
                 ""
                 isRegion
-                ""
-                ""]
+                src
+                dst]
 
 
 -- | Create an ellipse from an open ellipse tag.
 parseEllipse :: GraphId
              -> [Tag T.Text]
              -> [Shape]
 parseEllipse key tags =
-    map (parseEllipseHelper key trans) (filter (TS.isTagOpenName "ellipse") tags)
+    map (parseEllipseHelper key trans gid) (filter (TS.isTagOpenName "ellipse") tags)
     where
         trans = getTransform $ head tags
-
+        gid = (fromAttrib "id" $ head tags)
 
 parseEllipseHelper :: GraphId     -- ^ The related graph id.
                    -> Point       -- ^ The translation to apply.
+                   -> T.Text
```

Comment:
Add a comment here for the purpose of this parameter. (The `-- ^` makes the description picked up by Haddock.)

Reply:
done. Added small description